collectors/build/sonar/src/main/java/com/capitalone/dashboard/collector/DefaultSonar6Client.java

```diff
@@ -154,6 +159,54 @@ public CodeQuality currentCodeQuality(SonarProject project, String metrics) {
 
         return null;
     }
+    
+    public List<String> retrieveProfileAndProjectAssociation(String instanceUrl,String qualityProfile){
+    	List<String> projects = new ArrayList<>();
+    	String url = instanceUrl + URL_QUALITY_PROFILE_PROJECT_DETAILS + qualityProfile;
+    	try {
+    		JSONArray associatedProjects = this.parseAsArray(url, "results");
+    		if (associatedProjects != null) {
+    			for (Object project : associatedProjects) {
+    				JSONObject projectJson = (JSONObject) project;
+    				String projectName = (String) projectJson.get("name");
+    				projects.add(projectName);
+    			}
+    			return projects;
+    		}
+    		return null;
+    	} catch (ParseException e) {
```

Comment:
I think better design of these is to throw back the exceptions to the callers instead of catching and returning null.

Reply:
Done in the latest commit